apps/desktop-wallet/src/storage/addresses/addressesActions.ts

```
@@ -63,27 +63,23 @@ export const addressSettingsSaved = createAction<{ addressHash: AddressHash; set
   'addresses/addressSettingsSaved'
 )
 
-export const syncAddressesData = createAsyncThunk<
-  AddressTransactionsSyncResult[],
-  AddressHash[] | undefined,
-  { rejectValue: SnackbarMessage }
->('addresses/syncAddressesData', async (payload, { getState, dispatch, rejectWithValue }) => {
-  dispatch(syncingAddressDataStarted())
-
-  const state = getState() as RootState
-  const addresses = payload ?? (state.addresses.ids as AddressHash[])
-
-  try {
-    await dispatch(syncAddressesBalances(addresses))
-    await dispatch(syncAddressesTokensBalances(addresses))
-    return await dispatch(syncAddressesTransactions(addresses)).unwrap()
-  } catch (e) {
-    return rejectWithValue({
-      text: getHumanReadableError(e, i18n.t("Encountered error while syncing your addresses' data.")),
-      type: 'alert'
-    })
+export const syncAddressesData = createAsyncThunk(
+  'addresses/syncAddressesData',
+  async (payload: AddressHash[] | undefined, { getState, dispatch }) => {
+    dispatch(syncingAddressDataStarted())
+
+    const state = getState() as RootState
+    const addresses = payload ?? (state.addresses.ids as AddressHash[])
+
+    try {
+      await dispatch(syncAddressesBalances(addresses))
+      await dispatch(syncAddressesTokensBalances(addresses))
+      await dispatch(syncAddressesTransactions(addresses))
+    } catch (e) {
+      throw getHumanReadableError(e, i18n.t("Encountered error while syncing your addresses' data."))
```

Throwing instead of returning, to improve error logging